homeassistant/components/reolink/host.py

```diff
@@ -435,6 +435,11 @@ async def renew(self) -> None:
 
     async def _renew(self, sub_type: SubType) -> None:
         """Execute the renew of the subscription."""
+        if sub_type == SubType.all:
+            await self._renew(SubType.push)
```

Comment:
This looks like more than a typing change. If it's a bug fix, please solve it separately.

Reply:
Well, `_renew `is never called with argument `SubType.all`, but in the theoritical case it would, the underlying function in the upstream lib has input argument Literal[SubType.push] | Literal[SubType.long_poll], so mypy was complaining it could not deal with `SubType.all`.

In first instance I fixed that by using _renew(self, sub_type: Literal[SubType.push] | Literal[SubType.long_poll]), which on my local dev enviroment was fine and mypy passed.
But running here on github it gave this error, see https://github.com/home-assistant/core/actions/runs/6708573933/job/18229996254:
`error: Parameter 1 of Literal[...] cannot be of type "Any"  [valid-type]`

Honestly I do not understand why it gives that error on github on not on my local dev.....
The error also does not make sense to me the argument of `Literal` was `SubType.push` which is an value of a string Enum...

So this was the way I fixed it.

 @MartinHjelmare Do you know why `Literal[SubType.push] | Literal[SubType.long_poll]` would not work on github mypy?

Reply:
Maybe because the library isn't typed yet with `py.typed`?

Reply:
If it can't happen we shouldn't have code doing things for that case.

Reply:
You are a genious, of course!
In my local dev I indeed already have the py.typed in the reolink-aio lib while on github it is not there.
(still need to release new version of reolink-aio).

Will make the changes!